Snakefile

```diff
@@ -275,6 +275,18 @@ PARALLEL_LANGUAGE_PAIRS = [
     'en_ru', 'en_sv', 'en_zh-Hans', 'en_zh-Hant'
 ]
 
+# We want some flexibility in the amount of data we encode via SentencePiece
+# to use in training a language model.  That data will be a subset of the
+# monolingual data sampled to train SentencePiece itself; we define the size
+# of that subset here as the reciprocal of the fraction of the whole monolingual
+# dataset to use.  (So setting this to, say, 5 uses one part in 5 of the data.)
+RECIPROCAL_OF_FRACTION_OF_DATA_TO_SPM_ENCODE = 5
+
+# To have a (statically) predictable set of SentencePiece id files when
+# segregated by length (number of id's per input, not the number of inputs),
+# we set an upper bound on the length.
+MAX_SPM_ENCODE_IDS = 300
```

Comment:
The upper bound on the length still needs to be much lower than 300. That is too many tokens for an LSTM model on a consumer GPU to back-propagate through.

Reply:
Changed to 70.